### PR TITLE
Fixed replay, added checking integrity

### PR DESCRIPTION
**Pull request by Andrei-Masilevich**

### CODE REVIEW

File: libraries/chain/block_log.cpp [hunk 1]
```diff
@@ -184,6 +184,11 @@ bool block_log::is_open() const
     return my->block_stream.is_open();
 }
 
+fc::path block_log::block_log_index_path(const fc::path& file)
```
Comment (alexplevako) [inline libraries/chain/block_log.cpp]: const

File: libraries/chain/database/database.cpp [hunk 2]
```diff
@@ -115,6 +117,11 @@ database::~database()
     clear_pending();
 }
 
+fc::path database::block_log_path(const fc::path& data_dir)
```
Comment (alexplevako) [inline libraries/chain/database/database.cpp]: const

Comment (Andrei-Masilevich) [reply libraries/chain/database/database.cpp]: static???

File: libraries/chain/database/database.cpp [hunk 3]
```diff
@@ -210,15 +221,15 @@ void database::reindex(const fc::path& data_dir,
...
-            auto last_block_num = _block_log.head()->block_num();
-
             while (itr.first.block_num() != last_block_num)
             {
                 auto cur_block_num = itr.first.block_num();
-                if (cur_block_num % 100000 == 0)
-                    std::cerr << "   " << double(cur_block_num * 100) / last_block_num << "%   " << cur_block_num
-                              << " of " << last_block_num << "   (" << (get_free_memory() / (1024 * 1024))
-                              << "M free)\n";
+                if (cur_block_num % log_interval_sz == 0)
```
Comment (Katkevich) [inline libraries/chain/database/database.cpp]: Add ` || cur_block_num == last_block_num` condition here so we log info for the rest of blocks

**Pull request merged**